python/cudf/tests/test_generic_ptx.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+from __future__ import division
+
+import pytest
+import numpy as np
+
+from cudf.bindings import binops
+from cudf.dataframe import Series
+
+from numba import cuda
+from numba import types
+
+
+@pytest.mark.parametrize('dtype', ['int32', 'int64', 'float32', 'float64'])
+def test_generic_ptx(dtype):
+
+    size = 500
+
+    lhs_arr = np.random.random(size).astype(dtype)
+    lhs_col = Series(lhs_arr)._column
+
+    rhs_arr = np.random.random(size).astype(dtype)
+    rhs_col = Series(rhs_arr)._column
+
+    out_arr = np.random.random(size).astype(dtype)
+    out_col = Series(out_arr)._column
+
+    @cuda.jit(device=True)
+    def add(a, b):
+        return a**3 + b
+    if dtype == 'float32':
```

Comment:
If it's not part of the `add` function, I really think you should have a blank line after the function definition. (Since Python lacks grouping symbols).

```suggestion

    if dtype == 'float32':
```

Reply:
fixed that.